src/components/mobile/survey-questions.tsx

```diff
@@ -82,6 +91,83 @@ export function MobileSurveyQuestionnaire({
     },
   });
 
+  useEffect(() => {
```

Comment:
In general, this effect runs far too long. In general, try to not use `useEffect` directly from a component, instead move it to helper methods.